tests/test_authority.py

```diff
@@ -186,6 +186,28 @@ def test_url_extra_path_elements(self):
                                                      "https://login.microsoftonline.com/your_tenant"):
             context = AuthenticationContext(self.nonHardCodedAuthority + '/extra/path')
 
+    @httpretty.activate
+    def test_url_extra_slashes(self):
+        util.setup_expected_instance_discovery_request(200,
+                                                       cp['authorityHosts']['global'],
+                                                       {
+                                                           'tenant_discovery_endpoint': 'http://foobar'
+                                                       },
+                                                       self.nonHardCodedAuthorizeEndpoint)
+
+        authority_url = self.nonHardCodedAuthority + '//'
+        authority = Authority(authority_url, True)
+        obj = util.create_empty_adal_object()
+        authority.validate(obj['call_context'])
+        req = httpretty.last_request()
+        util.match_standard_request_headers(req)
+
+    @httpretty.activate
+    def test_url_extra_slashes_change_authority_url(self):
+        authority_url = self.nonHardCodedAuthority + '//'
```

Comment:
The number of slashes here actually don't matter because any number of slashes should be stripped. It just makes sure any number of slashes after the authority url are stripped before going further.

Reply:
I can change it real quick 👍

Reply:
why not have test for both? 

---
In reply to: [194158095](https://github.com/AzureAD/azure-activedirectory-library-for-python/pull/157#discussion_r194158095) [](ancestors = 194158095)

Reply:
What if we add a comment in the existing one which explains that it is designed to handle one or more slashes?

Reply:
tests should cover both good and bad scenarios. it's not all but ensuring we cover test cases ensuring we do not break the behavior we introduced (namely stripping of all / and ensuring there is always only 1 when we create the url.

Reply:
Aside: I already gave sign-off.